db/mongodb.go

```diff
@@ -0,0 +1,170 @@
+package db
+
+import (
+	"encoding/json"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+	"time"
+)
+
+var (
+	cronCollection = "cronjobs"
+)
+
+type mongoCronJob struct {
+	ID       bson.ObjectId `bson:"_id,omitempty"`
+	IsActive bool          `bson:"active"`
+	Function string        `bson:"function"`
+	Workload interface{}   `bson:"workload"`
+	CronTime string        `bson:"time"`
+	TimeZone string        `bson:"tz"`
+	Created  time.Time     `bson:"created"`
+}
+
+func (c CronJob) toMongoCronJob() mongoCronJob {
+	return mongoCronJob{
+		ID:       bson.ObjectIdHex(c.ID),
+		IsActive: c.IsActive,
+		Function: c.Function,
+		Workload: parseWorkload(c.Workload),
+		CronTime: c.CronTime,
+		TimeZone: c.TimeZone,
+		Created:  c.Created,
+	}
+}
+
+func (mc *mongoCronJob) toCronJob() CronJob {
+	var workload string
+	switch t := mc.Workload.(type) {
+	case string:
+		workload = t
+	case bson.M:
+		workloadByte, _ := json.Marshal(t)
+		workload = string(workloadByte)
+	}
+
+	return CronJob{
+		ID:       mc.ID.Hex(),
+		IsActive: mc.IsActive,
+		Function: mc.Function,
+		Workload: workload,
+		CronTime: mc.CronTime,
+		TimeZone: mc.TimeZone,
+		Created:  mc.Created,
+	}
+}
+
+// MongoDB is the visible DB interface used in server
+type MongoDB struct {
+	Session *mgo.Session
+	DBName  string
+}
+
+func (db *MongoDB) sessionClone() *mgo.Session {
+	return db.Session.Clone()
+}
+
+func (db *MongoDB) getCronCollection(session *mgo.Session) *mgo.Collection {
+	return session.DB(db.DBName).C(cronCollection)
+}
+
+// NewMongoDB attempts to connect with DB and errs if problems are found
+func NewMongoDB(mongoURL string, dbName string) (*MongoDB, error) {
+	session, dialErr := mgo.Dial(mongoURL)
+	if dialErr != nil {
+		return nil, dialErr
+	}
+	return &MongoDB{Session: session, DBName: dbName}, nil
+}
+
+// GetDistinctActiveFunctions returns a list of functions with active jobs
+func (db *MongoDB) GetDistinctActiveFunctions() ([]string, error) {
+	var activeFunctions []string
+
+	session := db.sessionClone()
+	defer session.Close()
+	collection := db.getCronCollection(session)
+
+	query := collection.Find(bson.M{"active": true})
+	if err := query.Distinct("function", &activeFunctions); err != nil {
+		return activeFunctions, err
+	}
+	return activeFunctions, nil
+}
+
+// GetJobDetails returns all jobs associated with a function
+func (db *MongoDB) GetJobDetails(function string) ([]CronJob, error) {
+	var mongoJobDetails []mongoCronJob
+	var jobDetails []CronJob
+
+	session := db.sessionClone()
+	defer session.Close()
+	collection := db.getCronCollection(session)
+
+	query := collection.Find(bson.M{"function": function})
+	if err := query.All(&mongoJobDetails); err != nil {
```

Comment:
what happens if the `Find` returns zero results? I think `mgo` returns `mgo.ErrNotFound`, in which case your error handling here might need to change.

Might be a good place to start testing :)